rca/api_content/content.py

```diff
@@ -0,0 +1,132 @@
+        if type == "Event":
+            detail = item["meta"]["detail_url"] + "?fields=_,dates_times,feed_image"
+        if type == "Alumni stories":
+            detail = item["meta"]["detail_url"] + "?fields=_,feed_image,intro"
+
+        try:
+            response = requests.get(url=detail)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
+            logger.exception("Error occured when fetching further detail data")
+
+        data = response.json()
+        if "feed_image" in data:
+            feed_image = data["feed_image"]["meta"]["detail_url"]
+            feed_image = requests.get(url=feed_image)
+            feed_image = feed_image.json()
+            feed_image_url = feed_image["original"]["url"]
+            _item["image"] = feed_image_url
+            _item["image_alt"] = feed_image["alt"]
+        date = False
+        if type == "News":
+            date = data["date"]
+        if type == "Event":
+            date = data["dates_times"][0]["date_from"]
+        if type == "Alumni stories":
+            _item["description"] = data["intro"]
+        if date:
+            date = datetime.strptime(date, "%Y-%m-%d")
+            date = date.strftime("%-d %B %Y")
+            _item["description"] = date
+
+        _item["title"] = item["title"]
+        _item["type"] = type
+        _item["link"] = item["meta"]["html_url"]
+        items.append(_item)
+
+    return items
+
+
+def pull_news_and_events(programme_type_slug=None):
+    # Get the latest 2 news items and 1 event item tagged
+    # with the programme type taxonomy, if there are no events, use 3 news
+    query = QueryDict(mutable=True)
+    query.update({"limit": 1, "event_date_from": True, "type": "rca.EventItem"})
+
+    if programme_type_slug:
+        query.update({"rp": programme_type_slug})
+
+    query = query.urlencode()
+    events_url = f"https://rca.ac.uk/api/v2/pages/?{query}"
+
+    # First get the events
+    try:
+        response = requests.get(url=events_url)
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.exception("Error occured when fetching events data")
+
+    data = response.json()
+    events_data = []
+    if data["meta"]["total_count"] > 0:
+        news_items_to_get = 2
+        events_data = parse_items_to_list(data, "Event")
+    else:
+        news_items_to_get = 3
+
+    # News
+    query = QueryDict(mutable=True)
+    query.update({"limit": news_items_to_get, "order": "-date", "type": "rca.NewsItem"})
+    if programme_type_slug:
+        query.update({"rp": programme_type_slug})
+    query = query.urlencode()
+    news_url = f"https://rca.ac.uk/api/v2/pages/?{query}"
+
+    try:
+        response = requests.get(url=news_url)
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.exception("Error occured when fetching news data")
+
+    data = response.json()
```

Comment:
I do not think you want to call this after you had a HTTP error. You want to use `else` block or put it in the `try` block.

Reply:
I've improved the exception handling so this is ready for re-review, thanks.